src/index.jsx

```diff
@@ -1,19 +1,31 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
+import { addLocaleData, IntlProvider } from 'react-intl';
+import enLocaleData from 'react-intl/locale-data/en';
+import frLocaleData from 'react-intl/locale-data/fr';
 
 import AssetsPage from './AssetsPage';
 import BackendStatusBanner from './BackendStatusBanner';
 import store from './data/store';
 
+import frenchMessages from '../i18n/locales/fr.json';
+
+// en is hardcoded in right now, eventually this will be a dynamic locale
+addLocaleData(enLocaleData);
+addLocaleData(frLocaleData);
```

Comment:
This is going to be the difficult portion to solve. When the locale is hardcoded, I can easily import the correct messages (line 12) and `addLocaleData` with the correct data (required by `IntlProvider`, something to do with knowing how to format dates I think?).

But how will we be able to do this on-the-fly? I'm very open to ideas here - keep in mind that we *only* want to import `*LocaleData` and messages for the current language, not all of them.

Could we give each locale a small script, located at `../i18n/locales/<locale>.js` to load localedata and messages? Then we can (in platform) give each locale a webpacked entry point, and dynamically include it into the template at render time...

Reply:
Yes, very tricky!

Just to check my understanding, let me restate. If we include each and every local data that we have, our webpack-ed bundle would grow as we add languages, pages, and as translations come in. We'd rather have *only* the correct language JSON loaded. Therefore, we don't want the language bundled.

Because the platform manages the bundle, we'd need to find a way for the language data to be injected in. Could we attach it to the `root` in a `data-*` attribute and then load it here?